src/FinalFrame.js

```diff
@@ -0,0 +1,22 @@
+class FinalFrame extends Frame {
+  constructor() {
+    super();
+    this.rollThree = 0;
+  }
+
+  roll(value) {
+    if (this.currentroll === 1) {
+      this.rollOne = value;
+      this.currentroll++;
+    } else if (this.currentroll === 2) {
+      this.rollTwo = value;
+      this.currentroll++;
+      if (!this.isStrike() || !this.isSpare()) {
+        this.endFrame = true;
+      }
+    } else if (this.isStrike() || this.isSpare()) {
+      this.rollThree = value;
+      this.endFrame = true;
```

Comment:
May want to look at refactoring large else if statements to a switch statement OR extract some of the code into a separate method?